### PR TITLE
HEP 578 event and audit log [CI SKIP]

### PR DESCRIPTION
**Pull request by w13915984028**
Signed-off-by: Jian Wang <w13915984028@gmail.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

Add event and audit logs.

**Solution:**

Per HEP doc.
https://github.com/w13915984028/harvester/blob/hep578/enhancements/20220822-event-and-audit.md

**Related Issue:**
https://github.com/harvester/harvester/issues/578

**Test plan:**

### CODE REVIEW

**Comment by w13915984028** [comment]
@guangbochen @bk201 @joshmeranda @yasker  @DaiYuzeng 

Ready for review and meeting discussion.

File: enhancements/20220822-event-and-audit.md [hunk 1]
```diff
@@ -0,0 +1,709 @@
+# Title
+
+Event and audit log
```
Comment (bk201) [inline enhancements/20220822-event-and-audit.md]: nit: I guess this should be moved to the `# Title`

File: enhancements/20220822-event-and-audit.md [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+
+Events have no direct relationship with log messages generated by the various components, and are not affected with the log verbosity level. When a component creates an event, it often emits a corresponding log message. Events are garbage collected by the API Server after a short time (typically after an hour, 5 minutes in case of EKS), which means that they can be used to understand issues just happening, but you have to collect them to investigate past events.
+
+Events are the first thing to look at for application, as well as infrastructure operations when something is not working as expected. Keeping them for a longer period is essential if the failure is the result of earlier events, or when conducting post-mortem analysis.
+
+### Goals
+
+Collect `audit` log by default.
+Collect `event` log by default.
+Send `audit` and `event` log to `loki` by default, which can be queried by embedded `grafana`.
```
Comment (bk201) [inline enhancements/20220822-event-and-audit.md]: The loki is not enabled by default, need to revise. Users can send logs to their data platform or provision a loki deployment.

File: enhancements/20220822-event-and-audit.md (hunk 1 above)
Comment (w13915984028) [reply enhancements/20220822-event-and-audit.md]: updated, thanks.

File: enhancements/20220822-event-and-audit.md (hunk 2 above)
Comment (w13915984028) [reply enhancements/20220822-event-and-audit.md]: `loki` related parts are all moved to a separate chapter, thanks.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]
LGTM

**Pull request merged**